CRM/Sepa/Logic/Queue/Update.php

```diff
@@ -23,121 +23,182 @@
  */
 class CRM_Sepa_Logic_Queue_Update {
 
-  public $title          = NULL;
-  protected $cmd         = NULL;
-  protected $mode        = NULL;
-  protected $creditor_id = NULL;
-  protected $offset      = NULL;
-  protected $limit       = NULL;
+  /**
+   * Create a task for the queue
+   *
+   * @param $cmd string
+   * @param $params array
+   * @return CRM_Queue_Task
+   */
+  private static function createTask($cmd, $params = []) {
+    $task = new CRM_Queue_Task(
+      ['CRM_Sepa_Logic_Queue_Update', 'run'],
+      [$cmd, $params],
+      self::taskTitle($cmd, $params)
+    );
+
+    $task->runAs = [
+      'contactId' => CRM_Core_Session::getLoggedInContactID(),
+      'domainId'  => 1,
+    ];
+
+    return $task;
+  }
 
   /**
-   * Use CRM_Queue_Runner to do the SDD group update
-   * This doesn't return, but redirects to the runner
+   * determine the count of mandates to be investigated
+   */
+  protected static function getMandateCount($creditor_id, $sdd_mode) {
```

Comment:
Just noting that we should think about whether the fact that `UPDATE` now may run in parallel has any effect on the offset-based batching approach used here.